packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_list.dart

```diff
@@ -125,21 +125,24 @@ class _SnapshotListItemsState extends State<_SnapshotListItems>
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    addAutoDisposeListener(widget.controller.selectedIndex, scrollIfLast);
+    addAutoDisposeListener(
+      widget.controller.selectedSnapshotIndex,
+      scrollIfLast,
+    );
```

Comment:
this can be moved to initState, and needs to be called in didUpdateWidget if `widget.controller != oldWidget.controller`